autobotAI_integrations/integrations/ms_teams/__init__.py

```diff
@@ -0,0 +1,152 @@
+from typing import Type, Union
+
+import uuid
+import re, pymsteams
+
+from autobotAI_integrations import BaseService, list_of_unique_elements, PayloadTask, Param
+from autobotAI_integrations.models import *
+import importlib
+
+
+class MsTeamsIntegration(BaseSchema):
+    webhook: str
+    accountId: Optional[str] = None
+
+    def __init__(self, **kwargs):
+        kwargs["accountId"] = str(uuid.uuid4().hex)
+        super().__init__(**kwargs)
+
+
+class MsTeamsService(BaseService):
+    def __init__(self, ctx: dict, integration: Union[MsTeamsIntegration, dict]):
+        """
+        Integration should have all the data regarding the integration
+        """
+        if not isinstance(integration, MsTeamsIntegration):
+            integration = MsTeamsIntegration(**integration)
+        super().__init__(ctx, integration)
+
+    def _test_integration(self, integration: dict) -> dict:
+        res = None
+        pattern = re.compile(
+            "https:\\/\\/[\\w\\-\\.]+\\/webhookb2\\/[\\w\\d\\-\\@]+\\/IncomingWebhook\\/[\\w\\d\\-\\@]+\\/[\\w\\d\\-\\@]+")
+        result = pattern.match(integration["webhook"])
+        if result is None:
+            res = {'success': False, "error": "Webhook is not valid ms_teams webhook URL"}
+        try:
+            if not res:
+                client = pymsteams.connectorcard(integration["webhook"])
+                client.title("Teams Integration Test Title")
+                client.text("Teams Integration Test Body")
+                client.send()
+                res = {'success': True}
+        except:
+            res = {'success': False, "error": "Unable to send message to Webhook"}
+
+    @staticmethod
+    def get_forms():
+        return {
+            "label": "Azure",
+            "type": "form",
+            "children": [
+                {
+                    "label": "MS Teams",
+                    "type": "form",
+                    "children": [
+                        {
+                            "name": "webhook",
+                            "type": "text/url",
+                            "label": "Webhook URL",
+                            "placeholder": "Enter your Webhook URL",
+                            "required": True
+                        }
+                    ]
+                }
+            ]
+        }
+
+
+    @staticmethod
+    def get_schema() -> Type[BaseSchema]:
+        return MsTeamsIntegration
+
+    @classmethod
+    def get_details(cls):
+        return {
+            "fetcher_supported": [],
+            "listener_supported": False,
+            "automation_supported": ['communication'],
+            "clients": list_of_unique_elements(MsTeamsService.get_clients()),
```

Comment:
This will return error as there is no method called get_clients in MsTeamsService